libs/cardano-ledger-binary/testlib/Test/Cardano/Ledger/Binary/RoundTrip.hs

```diff
@@ -83,30 +82,29 @@ roundTripFailureExpectation version x =
       expectationFailure $
         "Should not have deserialized: " ++ showExpr (CBORBytes (serialize' version x))
 
--- | Verify that round triping through the binary form holds.
+-- | Verify that round triping through the binary form holds fro all possible versions
 --
 -- In other words check that:
 --
 -- > deserialize version . serialize version === id
 -- > serialize version . deserialize version . serialize version === serialize version
 roundTripExpectation ::
   (Show t, Eq t, Typeable t, HasCallStack) =>
-  Version ->
   Trip t t ->
   t ->
   Expectation
-roundTripExpectation version trip t =
-  case roundTrip version trip t of
-    Left err -> expectationFailure $ "Failed to deserialize encoded: " ++ show err
-    Right tDecoded -> tDecoded `shouldBe` t
+roundTripExpectation trip t =
+  forM_ [minBound .. maxBound] $ \version ->
+    case roundTrip version trip t of
+      Left err -> expectationFailure $ "Failed to deserialize encoded: " ++ show err
```

Comment:
Does `err` include the version number for which it failed to deserialize?

Reply:
Yep, that was the part I showed during the meeting. The roundTrip produces failure result that contains the version, along with other relevant stuff